packages/wonder-blocks-announcer/src/Announcer.ts

```diff
@@ -0,0 +1,291 @@
+        this.node.id = id;
+        this.node.setAttribute("data-testid", id);
+
+        Object.assign(this.node.style, srOnly);
+
+        const aWrapper = this.createRegionWrapper("assertive");
+        this.createDuplicateRegions(aWrapper, "assertive");
+        this.node?.appendChild(aWrapper);
+
+        const pWrapper = this.createRegionWrapper("polite");
+        this.createDuplicateRegions(pWrapper, "polite");
+        this.node.appendChild(pWrapper);
+
+        document.body.prepend(this.node);
+    }
+    /**
+     * Recover in the event regions get lost
+     * This happens in Storybook when saving a file:
+     * Announcer exists, but it loses the connection to element Refs
+     */
+    rebootForHMR() {
+        const announcerCheck = document.getElementById(`wbAnnounce`);
+        if (announcerCheck !== null) {
+            this.node = announcerCheck;
+            const regions = Array.from(
+                announcerCheck.querySelectorAll<HTMLElement>(
+                    "[id^='wbARegion'",
+                ),
+            );
+            regions.forEach((region) => {
+                this.dictionary.set(region.id, {
+                    id: region.id,
+                    levelIndex: parseInt(
+                        region.id.charAt(region.id.length - 1),
+                    ),
+                    level: region.getAttribute("aria-live") as PolitenessLevel,
+                    element: region,
+                });
+            });
+        }
+    }
+
+    /**
+     * Create a wrapper element to group regions for a given level
+     * @param {string} level Politeness level for grouping
+     * @returns {HTMLElement} Wrapper DOM element reference
+     */
+    createRegionWrapper(level: PolitenessLevel) {
+        const wrapper = document.createElement("div");
+        wrapper.id = `wbAWrap-${level}`;
+        return wrapper;
+    }
+
+    /**
+     * Create multiple live regions for a given level
+     * @param {HTMLElement} wrapper Parent DOM element reference to append into
+     * @param {string} level Politeness level for grouping
+     * @returns {HTMLElement[]} Array of region elements
+     */
+    createDuplicateRegions(
+        wrapper: HTMLElement,
+        level: PolitenessLevel,
+    ): HTMLElement[] {
+        const result = new Array(this.regionFactory.count)
+            .fill(0)
+            .map((el, i) => {
+                const region = this.createRegion(level, i);
+                wrapper.appendChild(region);
+                return region;
+            });
+        return result;
+    }
+
+    /**
+     * Create live region element for a given level
+     * @param {string} level Politeness level for grouping
+     * @param {number} index Incrementor for duplicate regions
+     * @param {string} role Role attribute for live regions, defaults to log
+     * @returns {HTMLElement} DOM element reference for live region
+     */
+    createRegion(level: PolitenessLevel, index: number, role = "log") {
```

Comment:
What do you think about moving the document related utilities (`createRegion`, `createDuplicateRegions`, `createRegionWrapper` into a separate utility file? They could be tested on their own since they don't rely on the announcer instance! Then everything else in the `Announcer` class can be around managing announcements without worrying how to update the DOM!

Reply:
Good idea! I'll see what I can do with those...